### ISSUE TITLE
Issue discovered with `src/test/java/com/microsoft/jenkins/appservice/test/FilePathUtilsTest.java`

### ISSUE THREAD

**Issue by jkschneider** [openrewrite/rewrite-testing-frameworks#242]
### Problem
> Describe the issue you are experiencing.

### Expected behavior
> Describe what you expected to see.

### Example diff

```diff
 import hudson.FilePath;
 import org.apache.commons.io.FilenameUtils;
 import com.microsoft.jenkins.appservice.util.FilePathUtils;
-import org.junit.Assert;
 import org.junit.Rule;
-import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
 import org.junit.rules.TemporaryFolder;
 
 public class FilePathUtilsTest {
```

```diff
     public TemporaryFolder workspaceDir = new TemporaryFolder();
 
     @Test
-    public void trimDirectoryPrefix() throws Exception {
-        assertTrimDirectoryPrefix("a", "a/b/c.txt", "b/c.txt");
+    void trimDirectoryPrefix() throws Exception {
+        /*~~(java.lang.NullPointerException: Cannot invoke "org.openrewrite.java.tree.JavaType$FullyQualified.getFullyQualifiedName()" because the return value of "org.openrewrite.java.tree.JavaType$Method.getDeclaringType()" is null
+  org.openrewrite.java.testing.junit5.TemporaryFolderToTempDir$TemporaryFolderToTempDirVisitor.visitMethodInvocation(TemporaryFolderToTempDir.java:112)
+  org.openrewrite.java.testing.junit5.TemporaryFolderToTempDir$TemporaryFolderToTempDirVisitor.visitMethodInvocation(TemporaryFolderToTempDir.java:67)
+  org.openrewrite.java.tree.J$MethodInvocation.acceptJava(J.java:3409)
+  org.openrewrite.java.tree.J.accept(J.java:55)
+  org.openrewrite.TreeVisitor.visit(TreeVisitor.java:206)
+  org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:285)
+  org.openrewrite.java.JavaVisitor.visitRightPadded(JavaVisitor.java:1231)
+  org.openrewrite.java.JavaVisitor.lambda$visitBlock$4(JavaVisitor.java:373)
+  ...)~~>*/assertTrimDirectoryPrefix("a", "a/b/c.txt", "b/c.txt");
         assertTrimDirectoryPrefix("a", "a/b.txt", "b.txt");
         assertTrimDirectoryPrefix("a/", "a/b.txt", "b.txt");
         assertTrimDirectoryPrefix("", "c.txt", "c.txt");
```

```diff
         final FilePath file = workspace.child(filePath);
 
         final String remoteName = FilePathUtils.trimDirectoryPrefix(sourceDir, file);
-        Assert.assertEquals(expectedRemoteName, FilenameUtils.separatorsToUnix(remoteName));
+        Assertions.assertEquals(expectedRemoteName, FilenameUtils.separatorsToUnix(remoteName));
     }
 }
```

#### Recipes in example diff:
- org.openrewrite.java.testing.junit5.AssertToAssertions
- org.openrewrite.java.testing.junit5.TemporaryFolderToTempDir
- org.openrewrite.java.testing.junit5.UpdateTestAnnotation

### References:
- Recipe ID: `undefined`
- Recipe Name: "JUnit Jupiter migration from JUnit 4.x"
- Repository: microsoft/azure-app-service-plugin/dev
- Created at Wed Aug 03 2022 11:37:18 GMT-0400 (Eastern Daylight Time)

**Comment by pway99** [comment]
@jkschneider I was not able to reproduce this issue; however, the code should have been using TypeUtils.

The original code had a null check on `MethodType` and `MethodType.DeclaringType` is a non-null attribute, I am thinking its either an issue with the AST or the stack trace is not associated with the correct element.   The project was a bit challenging to build locally so I just updated the recipe and we can re-test on the SaaS.

```java
String declaringType = mi.getMethodType() != null ? mi.getMethodType().getDeclaringType().getFullyQualifiedName() : null;
if ("org.junit.rules.TemporaryFolder".equals(declaringType) && mi.getSelect() != null)
```

**Comment by pway99** [comment]
fixed in: 37fb56a149a25a2f45192688df699a93700ce8c2



![Image](https://user-images.githubusercontent.com/22326988/182708329-b1e30195-7e3d-4e2d-b00f-ffc606efa275.png)
